### PR TITLE
Add webauthn auth provider UI components

### PR DESCRIPTION
**Pull request by VDigitall**
## Proposed change

Add webauthn auth provider UI functionality.

Core PR: https://github.com/home-assistant/core/pull/122725

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
homeassistant:
  auth_providers:
    - type: homeassistant
    - type: webauthn
      rp_id: localhost
      expected_origin: http://localhost:8123
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/33998

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
	- Introduced support for WebAuthn authentication, enhancing security with passkey management.
	- Added a new interface for managing passkeys and related authentication processes.
	- Implemented a user-friendly component for viewing, adding, renaming, and deleting passkeys within user profiles.
	- Enhanced the authentication flow with improved error handling and dynamic UI updates based on authentication providers.

- **Bug Fixes**
	- Improved error handling for passkey operations to enhance user experience.

- **Documentation**
	- Updated translations to include new user interface elements and prompts related to passkeys and WebAuthn.

### CODE REVIEW

**Comment by c0ffeeca7** [comment]
I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

File: src/data/webauthn.ts
Comment (silamon) [inline src/data/webauthn.ts:140]:
```suggestion
export interface PublicKeyRegistrationCredentialResponseJSON {
```

File: src/data/webauthn.ts
Comment (silamon) [inline src/data/webauthn.ts:153]:
```suggestion
export interface PublicKeyRegistrationCredentialResponse {
```

**Review by silamon — COMMENTED** [review]
I've went through this amazing contribution. It may take some time for this feature to get approved in both frontend and core so I hope you are patient enough.

Here are some review comments:

**Comment by VDigitall** [comment]
@silamon Thanks for your review. All your comments were counted in the following commits.

**Comment by silamon** [comment]
The failing lint isn't related to your pull request, but I'm not sure why it pops up here for the first time either. Anyway, those changes shouldn't be added to this branch.

**Comment by VDigitall** [comment]
Okay, I will remove last commit, thanks.

**Review by silamon — DISMISSED** [review]
Looks good to me, thank you for doing the changes. Let's wait for a second review and review in the backend code.

**Comment by silamon** [comment]
> Okay, I will remove the last commit, thanks.
> 
> P.S. Meanwhile, I pulled the current dev, ran lint there, and faced the same errors that CI

I've opened a separate pull request with the commit. It's the cache on dev being valid, hence not detecting the error.

**Pull request closed without merging**
